cognite_toolkit/_cdf_tk/client/api/lookup.py

```diff
@@ -249,7 +250,7 @@ def _lookup(self) -> None:
     def _id(self, external_id: SequenceNotStr[str]) -> dict[str, int]:
         if not self._has_looked_up:
             self._lookup()
-        return {external_id: self._cache[external_id] for external_id in external_id}
+        return {external_id: self._cache[external_id] for external_id in external_id if external_id in self._cache}
```

Comment:
I am not sure this is the correct solution. Seeing how this is used when `lookup.id(...)` is called it is assumed that each external ID will result in an internal ID. I think this is worth a discussion.